File: c/hamming/src/hamming.c
#include "hamming.h"
#include <string.h>

int compute(char *s1, char *s2)
{
  if (!s1 || !s2) return -1;
  int len = strlen(s1);
  if (len != (int)strlen(s2)) return -1;
  int distance = 0;

  for (int i = 0; i < len; i++)
  {
    if (s1[i] != s2[i])
    {
      distance++;
    }
  }
  return distance;
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
package Hamming

fun compute(strand1: String, strand2: String): Int {
  require(strand1.length == strand2.length) { "left and right strands must be of equal length." }
  return strand1.zip(strand2).count { (s1, s2) -> (s1 != s2) }
}